File: stage-common/src/main/java/io/github/solusmods/eternalcore/stage/impl/network/StagesNetwork.java
package io.github.solusmods.eternalcore.stage.impl.network;

import io.github.solusmods.eternalcore.network.api.util.NetworkUtils;
import io.github.solusmods.eternalcore.stage.impl.network.c2s.SyncStagesStoragePayload;
import lombok.AccessLevel;
import lombok.NoArgsConstructor;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class StagesNetwork {

    public static void init(){
        NetworkUtils.registerC2SPayload(SyncStagesStoragePayload.TYPE, SyncStagesStoragePayload.STREAM_CODEC, SyncStagesStoragePayload::handle);
    }
}

File: stage-common/src/main/kotlin/io/github/solusmods/eternalcore/stage/impl/network/StagesNetwork.kt
package io.github.solusmods.eternalcore.stage.impl.network

import dev.architectury.networking.NetworkManager
import io.github.solusmods.eternalcore.network.api.util.NetworkUtils.registerC2SPayload
import io.github.solusmods.eternalcore.stage.impl.network.c2s.RequestStageBreakthroughPacket
import io.github.solusmods.eternalcore.stage.impl.network.c2s.SyncStagesStoragePayload


object StagesNetwork {
    fun init() {
        registerC2SPayload<SyncStagesStoragePayload?>(
            SyncStagesStoragePayload.Companion.TYPE,
            SyncStagesStoragePayload.Companion.STREAM_CODEC
        ) { obj: SyncStagesStoragePayload?, context: NetworkManager.PacketContext? ->
            obj!!.handle(context!!)
        }
        registerC2SPayload<RequestStageBreakthroughPacket?>(
            RequestStageBreakthroughPacket.Companion.TYPE,
            RequestStageBreakthroughPacket.Companion.STREAM_CODEC
        ) { obj: RequestStageBreakthroughPacket?, context: NetworkManager.PacketContext? ->
            obj!!.handle(context!!)
        }
    }
}
